lib/base/process.cpp

```diff
@@ -1129,7 +1129,7 @@ bool Process::DoEvents()
 	m_ResultCondition.notify_all();
 
 	if (m_Callback)
-		Utility::QueueAsyncCallback(std::bind(m_Callback, m_Result));
+		Utility::QueueAsyncCallback([this]() { return m_Callback(m_Result); });
```

Comment:
Explain with your own words why this will crash.

Reply:
Missing forward declared return type.

Reply:
Are the things you use in the lambda captured by value or reference?

Reply:
Of course by value because I capture the ``this`` keyword and this cannot be captured by reference.

Reply:
Ok, _the this pointer_ is captured by value. And m_Result?

Reply:
Since I am only accessing the variable using this keyword, I assume it is also by value and not by reference.

> According to the name of the method, it is an async callback. Would it be appropriate to make this `this` keyword reference counted?

Here also the same question?

Reply:
I think I don't even need to return a value here because ``m_Callback`` a void pointer.

Reply:
> Here also the same question?

And [the same](https://github.com/Icinga/icinga2/pull/8601#discussion_r560949093) answers.